src/AWS/S3/S3.purs

```diff
@@ -0,0 +1,130 @@
+module AWS.S3
+  ( S3
+  , BucketName
+  , BucketPolicyParams
+  , BucketPolicy
+  , createBucket
+  , CreateBucketResponse
+  , getObject
+  , GetObjectParams
+  , GetObjectResponse
+  , makeClient
+  , putBucketPolicy
+  ) where
+
+import AWS.Core.Client (makeClientHelper)
+import AWS.Core.Types (DefaultClientProps, Region(..))
+import Control.Promise (Promise, toAffE)
+import Data.Function.Uncurried (Fn2, runFn2, Fn3, runFn3)
+import Effect (Effect)
+import Effect.Aff (Aff)
+import Foreign (Foreign)
+import Justifill (justifillVia)
+import Justifill.Fillable (class Fillable)
+import Justifill.Justifiable (class Justifiable)
+import Node.Buffer (Buffer)
+import Prelude (Unit, ($), class Show, (#), (<#>))
+import Type.Proxy (Proxy(..))
+import Data.Newtype (class Newtype)
+
+foreign import data S3 :: Type
+
+foreign import newS3 :: Foreign -> Effect S3
+
+makeClient ::
+  forall r via.
+  Justifiable { | r } { | via } =>
+  Fillable { | via } DefaultClientProps =>
+  { | r } ->
+  Effect S3
+makeClient r = makeClientHelper newS3 props
+  where
+  viaProxy :: Proxy { | via }
+  viaProxy = Proxy
+
+  props :: DefaultClientProps
+  props = justifillVia viaProxy r
+
+type InternalGetObjectParams
+  = { "Bucket" :: String
+    , "Key" :: String
+    }
+
+type InternalGetObjectResponse
+  = { "Body" :: Buffer
+    , "ContentLength" :: Int
+    , "ContentEncoding" :: String
+    , "ContentType" :: String
+    }
+
+foreign import getObjectImpl :: Fn2 S3 InternalGetObjectParams (Effect (Promise InternalGetObjectResponse))
+
+type GetObjectParams
+  = { bucket :: String
+    , key :: String
+    }
+
+type GetObjectResponse
+  = { body :: Buffer
+    , contentLenght :: Int
+    , contentEncoding :: String
+    , contentType :: String
+    }
+
+getObject :: S3 -> GetObjectParams -> Aff GetObjectResponse
+getObject client input = runFn2 getObjectImpl client params # toAffE <#> convert
+  where
+  params =
+    { "Bucket": input.bucket
+    , "Key": input.key
+    }
+
+  convert :: InternalGetObjectResponse -> GetObjectResponse
+  convert internalResponse =
+    { body: internalResponse."Body"
+    , contentLenght: internalResponse."ContentLength"
+    , contentEncoding: internalResponse."ContentEncoding"
+    , contentType: internalResponse."ContentType"
+    }
+
+foreign import createBucketImpl :: Fn3 S3 String InternalBucketConfiguration (Effect (Promise Unit))
+
+type InternalBucketConfiguration
+  = { "LocationConstraint" :: String }
+
+type CreateBucketResponse
+  = { location :: String }
```

Comment:
What is location? Does it make sense to introduce a newtype or ADT?

Reply:
It could be `Region` actually (which we already have has a newtype in our Core folder).
But ideally this would be an ADT (see possible regions mention in the parameters of the createBucket function: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html).

Context: I've started to work on it for our "freezeLogs"-action in our CO project but put it on standby then to finish something else, that's why it's still a simple String there... 🤷‍♀️

Reply:
if location is Region then let's use the region newtype then. Agree on the ADT, but we can handle this in a different PR